src/definitions/modules/calendar.js

```diff
@@ -1073,6 +1125,44 @@ $.fn.calendar = function(parameters) {
             }
             return null;
           },
+          findHourAsObject: function(date, mode, hours) {
+            if (mode === 'hour') {
+              var d;
+              for (var i = 0; i < hours.length; i++) {
+                d = hours[i];
+                if (typeof d === 'number' && date.getHours() == d) {
+                  return null;
+                } else if (d !== null && typeof d === 'object') {
+                  if (d[metadata.days]) {
+                    if (typeof d[metadata.days] === 'number' && date.getDay() == d[metadata.days]) {
+                        return d;
+                    } else if (Array.isArray(d[metadata.days])) {
+                      if (d[metadata.days].indexOf(date.getDay()) > -1) {
+                        return d;
+                      }
+                    }
+                  } else if (d[metadata.date]) {
+                    if (d[metadata.date] instanceof Date && module.helper.dateEqual(date, module.helper.sanitiseDate(d[metadata.date]))) {
+                      return d;
+                    } else if (Array.isArray(d[metadata.date])) {
+                      if (d[metadata.date].some(function(idate) { return module.helper.dateEqual(date, idate, mode); })) {
+                        return d;
+                      }
+                    }
+                  } else if (d[metadata.hours]) {
+                    if (typeof d[metadata.hours] === 'number' && date.getHours() == d[metadata.hours]) {
+                      return d;
+                    } else if (Array.isArray(d[metadata.hours])) {
+                      if (d[metadata.hours].indexOf(date.getHours()) > -1) {
+                        return d;
+                      }
+                    }
```

Comment:
You have to check for the hours as well (as you did in the "is.disabled() method by determinging the "blocked" variable), otherwise the first found day will match.

I moved the last check for hours into a separate function to reuse the logic also when a day or date parameter is given

```suggestion
              var d;
              var hourCheck = function(date, d) {
                 if (d[metadata.hours]) {
                      if (typeof d[metadata.hours] === 'number' && date.getHours() == d[metadata.hours]) {
                     return d;
                  } else if (Array.isArray(d[metadata.hours])) {
                    if (d[metadata.hours].indexOf(date.getHours()) > -1) {
                      return d;
                    }
                  }
                }
              }
              for (var i = 0; i < hours.length; i++) {
                d = hours[i];
                if (typeof d === 'number' && date.getHours() == d) {
                  return null;
                } else if (d !== null && typeof d === 'object') {
                  if (d[metadata.days] && hourCheck(date,d)) {
                    if (typeof d[metadata.days] === 'number' && date.getDay() == d[metadata.days]) {
                        return d;
                    } else if (Array.isArray(d[metadata.days])) {
                      if (d[metadata.days].indexOf(date.getDay()) > -1) {
                        return d;
                      }
                    }
                  } else if (d[metadata.date] && hourCheck(date,d)) {
                    if (d[metadata.date] instanceof Date && module.helper.dateEqual(date, module.helper.sanitiseDate(d[metadata.date]))) {
                      return d;
                    } else if (Array.isArray(d[metadata.date])) {
                      if (d[metadata.date].some(function(idate) { return module.helper.dateEqual(date, idate, mode); })) {
                        return d;
                      }
                    }
                  } else if (hourCheck(date,d)) {
                 
```

Reply:
Changes done with 547d631. Again, thank you @lubber-de for finding the issue, I spent 3 hours yesterday trying to figure it out !